custom-vu/src/main/kotlin/jces1209/vu/page/PageLoader.kt

```diff
@@ -0,0 +1,81 @@
+package jces1209.vu.page
+
+import org.openqa.selenium.By
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions.or
+import org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated
+import com.atlassian.performance.tools.jiraactions.api.page.wait
+import org.openqa.selenium.support.ui.ExpectedCondition
+import java.time.Duration
+
+class PageLoader private constructor(
+    private val webDriver: WebDriver,
+    private val correctLocators: List<By>,
+    private val errorLocators: List<By>,
+    private val loadTimeout: Duration
+) {
+
+    fun waitForPageToLoad() {
+        webDriver.wait(
+            condition = (correctLocators + errorLocators).anyElementVisible(),
+            timeout = loadTimeout
+        )
+        assertNoErrors()
```

Comment:
I like the fact that you materialized the pattern into code. I have to think about it a little, but it looks very promising.

Reply:
I got that mostly from writing embeded C, it is common there to create macros in function that operaate on registers and stuff as:
`void doStuff( type1 val1, type2 val2) {`
`    assertParams(val1, val2);`
`    /*do stuff*/`
`    assertNoErrors();`
`    return;`